### PR TITLE
Shading improvements for generated GLSL

### PR DESCRIPTION
**Pull request by jstone-lucasfilm**
- Add a Fresnel model term to mx_environment_radiance, allowing for correct integration of conductor Fresnel on rough surfaces.  Thanks to Stephen Hill at Lucasfilm for catching this issue in the codebase!
- Unify epsilon tests for shading dot products in GLSL, addressing artifacts from backwards shading normals.
- Refactor shading logic for consistency and clarity.
- Add a pure conductor material to the test suite.

### CODE REVIEW

**Comment by jstone-lucasfilm** [comment]
Here's a visual comparison a metal material with high roughness, before and after these improvements to GLSL shading:

Original Shading:
RoughMetal_Original

Improved Shading:
RoughMetal_Improved

File: libraries/pbrlib/genglsl/mx_dielectric_brdf.glsl [hunk 1]
```diff
@@ -70,17 +64,21 @@ void mx_dielectric_brdf_indirect(vec3 V, float weight, vec3 tint, float ior, vec
         return;
     }
 
+    float NdotV = clamp(dot(N, V), M_FLOAT_EPS, 1.0);
+    if (NdotV == M_FLOAT_EPS)
+    {
+        weight = 0.0;
```
Comment (niklasharrysson) [inline libraries/pbrlib/genglsl/mx_dielectric_brdf.glsl:70]: Would it be more efficient to return base directly here?

```
result = base;
return;
```

File: libraries/pbrlib/genglsl/mx_conductor_brdf.glsl [hunk 2]
```diff
@@ -38,30 +32,24 @@ void mx_conductor_brdf_reflection(vec3 L, vec3 V, float weight, vec3 reflectivit
...
     float avgRoughness = mx_average_roughness(roughness);
-
-    vec3 Li = mx_environment_radiance(N, V, X, roughness, vec3(1.0), vec3(1.0), distribution);
-    vec3 F = mx_fresnel_conductor(dot(N, V), ior_n, ior_k);
-
+    vec3 F = mx_fresnel_conductor(NdotV, ior_n, ior_k);
     vec3 comp = mx_ggx_energy_compensation(NdotV, avgRoughness, F);
 
-    result = Li * F * comp * weight;
+    vec3 Li = mx_environment_radiance(N, V, X, roughness, vec3(1.0), vec3(1.0), ior_n, ior_k, distribution, 1);
```
Comment (niklasharrysson) [inline libraries/pbrlib/genglsl/mx_conductor_brdf.glsl:52]: For clarity maybe we could define "enum" constants for the `fresnelMode` values.

Comment (jstone-lucasfilm) [reply libraries/pbrlib/genglsl/mx_conductor_brdf.glsl:52]: That's a good idea, and I was thinking of making a change along those lines in an upcoming revision.

**Comment by jstone-lucasfilm** [comment]
For reference, here is a comparison of roughness wedges for Standard Surface Gold, showing the difference before and after these shading improvements:

Original Shading:
RoughnessWedge_Gold_EnergyCompensation_Table_01

Improved Shading:
RoughnessWedge_Gold_EnergyCompensation_Table_02

**Pull request merged**